File: python/334. Increasing Triplet Subsequence.py
#coding=utf-8
__author__ = 'xuxuan'
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        if len(nums)<3:
            return False
        arrays=[]
        for num in nums:
            l=len(arrays)
            if l==0:
                arrays.append(num)
            elif l==1:
                if num>arrays[0]:
                    arrays.append(num)
                else:
                    arrays=[num]
            elif l==2:
                if num!=arrays[0] and num!=arrays[1]:
                    if num>arrays[0] and num>arrays[1]:
                        arrays.append(num)
                    elif num>arrays[0] and num<arrays[1]:
                        arrays=[arrays[0],num]
                    elif num<arrays[0] and num>arrays[1]:
                        arrays=[arrays[1],num]
                    else:
                        arrays=[arrays[1],num]
            else:
                return True
        return len(arrays)==3

File: java/334. Increasing Triplet Subsequence.java
public class Solution {
    public boolean increasingTriplet(int[] nums) {
        if (nums.length<3) {
            return false;
        }
        List<Integer> list=new ArrayList<Integer>();
        list.add(nums[0]);
        for (int i = 1; i < nums.length; i++) {
            if (list.size()==1) {
                if (nums[i]<=list.get(0)) {
                    list.remove(0);
                }
                list.add(nums[i]);
            }else {
                if (nums[i]!=list.get(0)&&nums[i]!=list.get(1)) {
                    if (nums[i]>list.get(0)&&nums[i]>list.get(1)) {
                        return true;
                    }else if (nums[i]>list.get(0)&&nums[i]<list.get(1)) {
                        list.remove(1);
                    }else {
                        list.remove(0);
                    }
                    list.add(nums[i]);
                }
            }
        }
        return false;
    }
}